src/extensions/extra/HTMLHinter/errorDisplay.js

```diff
@@ -56,65 +104,165 @@ define(function (require, exports, module) {
+    //         return;
+    //     }
+    //     getCodeMirror().getDoc().addLineClass(errorObject.line, "background", "errorHighlight");
+    // }
 
     //Removes highlight from line in which error was present
-    function removeHighlight(line) {
-        if(!line) {
-            return;
-        }
-        getCodeMirror().getDoc().removeLineClass(line, "background", "errorHighlight");
-    }
+    // function removeLineHighlight(line) {
+    //     if(!line) {
+    //         return;
+    //     }
+    //     getCodeMirror().getDoc().removeLineClass(line, "background", "errorHighlight");
+    // }
 
     //Function that adds a button on the gutter (on given line nubmer) next to the line numbers
-    function showButton(errorObject){
+    function showButton(errorObject, animationType){
         getCodeMirror().addWidget(errorObject, errorToggle, false);
         $(errorToggle).attr("class", "hint-marker-positioning hint-marker-error").removeClass("hidden");
-        //Show tooltips message
-        $(".hint-marker-positioning").tooltipsy({content : "Click error icon for details", alignTo: "cursor", offset: [10, -10]});
+
+        // Sometimes we want this to pop in instead of appearing instantly,
+        // This is where that happens.
+        if(animationType == "animated") {
+            $(errorToggle).attr("class", "hint-marker-positioning hint-marker-error").addClass("pop");
+        }
+
+        // Show tooltips message
+        // TODO - this shows even when the error is open, so that's not great
+        // $(".hint-marker-positioning").tooltipsy({
+        //     content : "Click error icon for details",
+        //     alignTo: "cursor", // Can also use on an element
+        //     offset: [10, -10]
+        // });
     }
 
     // Function that removes gutter button
-    function removeButton(){
+    function removeButton(animationType){
         if(!errorToggle) {
             return;
         }
+
+        // Maybe sometimes we'll need to remove it instantly, so I'll add a speed argument
+        // for future use you fucken idiots.
+
+        var goodEmojis = [
+            "biceps","halo","heart","peace","sunglasses","wink","horns","thumbs"
+        ];
+
+        var randomEmoji = goodEmojis[Math.floor(Math.random() * goodEmojis.length)];
+
         if (errorToggle.parentNode) {
-            $(errorToggle).remove();
+
+            if(animationType == "instant") {
+                $(errorToggle).remove();
+            } else {
+                $(errorToggle).addClass("bye");
+
+                // Adds the class for the "positive" emoji we want to use
+                // partway through the animation. This is a workaround becuase we are
+                // using a CSS animation, so it's hard to make it dynamic.
+                setTimeout(function(el) {
+                    return function() {
+                        el.classList.add(randomEmoji);
+                    };
+                }(errorToggle), 400);
```

Comment:
Can you pull these numbers `400`, `1300` out and define them above as ALL_CAPS variables that are better documented?